Sources/ArrayExtensions.swift

```diff
@@ -196,6 +196,14 @@ extension Array where Element: Equatable {
     public func unique() -> Array {
         return reduce([]) { $0.contains($1) ? $0 : $0 + [$1] }
     }
+    
+    /// EZSE: Break array into arrays of desired step (inspired by ruby's `step`)
+    public func step(_ stepSize: Int) -> [[Element]]? {
```

Comment:
So it the array is k dimensional, and if a programmer performs step(n) on it, does it create n k-1 dimensional arrays in row echelon form? 

If that is indeed the case, we would require some test cases on multi dimensional arrays.